packages/dd-trace/src/appsec/iast/analyzers/unvalidated-redirect-analyzer.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+const InjectionAnalyzer = require('./injection-analyzer')
+const { UNVALIDATED_REDIRECT } = require('../vulnerabilities')
+const { getNodeModulesPaths } = require('../path-line')
+
+const EXCLUDED_PATHS = getNodeModulesPaths('express/lib/response.js')
+
+class UnvalidatedRedirectAnalyzer extends InjectionAnalyzer {
+  constructor () {
+    super(UNVALIDATED_REDIRECT)
+
+    this.addSub('datadog:http:server:response:set-header:finish', ({ name, value }) => this.analyze(name, value))
+  }
+
+  // TODO: In case the location header value is tainted, this analyzer should check the ranges of the tainted.
+  // And do not report a vulnerability if source of the ranges (range.iinfo.type) are exclusively url or path params
+  // to avoid false positives.
+  analyze (name, value) {
+    if (!this.isLocationHeader(name) || typeof value !== 'string') return
+
+    super.analyze(value)
+  }
+
+  isLocationHeader (name) {
+    return name && name.toLowerCase() === 'location'
```

Comment:
Can make sense if we also do a trim here? If the customer writes `'location '`, does the redirection work?
```suggestion
    return name && name.trim().toLowerCase() === 'location'
```

Reply:
For the record, headers like that are not valid and node throws a _Header name must be a valid HTTP token_ error.